File: buzzer.py
#include <stdio.h>
#include <pigpio.h>

int main(int argc, char *argv[])
{
    if (gpioInitialise() < 0)
    {
        fprintf(stderr, "pigpio initialisation failed\n");
        return 1;
    }

    /* Set GPIO modes */
    gpioSetMode(5, PI_OUTPUT);

    gpioWrite(5, 1); /* on */
    time_sleep(0.2);
    gpioWrite(5, 0); /* off */

    /* Stop DMA, release resources */
    gpioTerminate();

    return 0;
}

File: buzzer.c
#include <stdbool.h>
#include <pigpio.h>

#define BUZZER_GPIO 5
bool buzzerInitialised = false;

void init_buzzer(void) {
    gpioSetMode(BUZZER_GPIO, PI_OUTPUT);
    buzzerInitialised = true;
}

void buzz(void) {
    if (!buzzerInitialised)
        init_buzzer();

    gpioWrite(BUZZER_GPIO, 1); /* on */
}

void unbuzz(void) {
    if (!buzzerInitialised)
        init_buzzer();

    gpioWrite(BUZZER_GPIO, 0); /* off */
}

void cleanup_buzzer(void) {
    if (!buzzerInitialised) return;

    unbuzz();
    gpioSetMode(BUZZER_GPIO, PI_INPUT);

    buzzerInitialised = false;
}